README.md

```diff
@@ -0,0 +1,1116 @@
+# Cosmos-Hub Vega  Upgrade Test Instruction
+
+This document describes the  procedures to test cosmoshub Vega upgrade locally.
+This upgrade will bring the new release of Cosmos-SDK v0.44.0 and IBC 1.0-RC6 into gaia.
+
+## Version
+- presently running cosmoshub-4, Gaia version: v5.0.5
+- going to upgrade to cosmoshub-4, Gaia version: v6.0.0
+
+## Chain upgrade by cosmovisor
+
+### Test Plan
+This document uses the data exported from live cosmoshub-4 to mock the upgrade. We will run two nodes locally with an exported genesis file to upgrade both nodes to gaia v6.0.0-Vega by cosmovisor. One of the two nodes will be a validator with over 67% power.
+
+### Build the binary of old version
+```shell
+cd gaia
+git checkout release/v5.0.5
+make install
+```
+```shell
+gaiad version
+# v5.0.5
+```
+### Change the genesis file
+
+We have prepared a genesis file in this repo which was obtained by `gaiad export` on cosmoshub-4 network at height 7368387. Uncompress this genesis file and use it as the genesis data to mock the cosmoshub upgrade.
+
+```shell
+cd Vega-test
+gunzip genesis.json.gz
+# verify the hash
+cat genesis.json | shasum -a 256
+> 
+86f29f23f9df51f5c58cb2c2f95e263f96f123801fc9844765f98eca49fe188f
+```
+
+Change one validator in the genesis file to be a validator that you have the private key  and make this validator own over 67% power, so that you can start a local chain to produce blocks.
+
+#### Change the addresses and keys
+```shell
+# change chain id
+sed -i '' 's%"chain_id": "cosmoshub-4",%"chain_id": "test",%g' genesis.json
+
+# substitue "Certus One", this is our  node1, you can find the key info. in priv_validator_key_val1.json in this repo.
+sed -i '' 's%cOQZvh/h9ZioSeUMZB/1Vy1Xo5x2sjrVjlE/qHnYifM=%qwiUMxz3llsy45fPvM0a8+XQeAJLvrX3QAEJmRMEEoU=%g' genesis.json
+sed -i '' 's%B00A6323737F321EB0B8D59C6FD497A14B60938A%D5AB5E458FD9F9964EF50A80451B6F3922E6A4AA%g' genesis.json
+sed -i '' 's%cosmosvalcons1kq9xxgmn0uepav9c6kwxl4yh599kpyu28e7ee6%cosmosvalcons16k44u3v0m8uevnh4p2qy2xm08y3wdf92xsc3ve%g' genesis.json
+
+# substitue "Binance Staking", this is our node2, also the validator who will own over 67% power. you can find the key info. in priv_validator_key_val1.json in this repo. 
+# tendermint pub_key
+sed -i '' 's%W459Kbdx+LJQ7dLVASW6sAfdqWqNRSXnvc53r9aOx/o=%oi55Dw+JjLQc4u1WlAS3FsGwh5fd5/N5cP3VOLnZ/H0=%g' genesis.json
+# priv_val address
+sed -i '' 's%83F47D7747B0F633A6BA0DF49B7DCF61F90AA1B0%7CB07B94FD743E2A8520C2B50DA4B03740643BF5%g' genesis.json
+#  Validator consensus address, try command ` gaiad keys parse 83F47D7747B0F633A6BA0DF49B7DCF61F90AA1B0` to see if you can get the same addr.
+sed -i '' 's%cosmosvalcons1s0686a68krmr8f46ph6fklw0v8us4gdsm7nhz3%cosmosvalcons10jc8h98awslz4pfqc26smf9sxaqxgwl4x`vxpcrp%g' genesis.json
```

Comment:
There's a stray character introduced here cosmosvalcons10jc8h98awslz4pfqc26smf9sxaqxgwl4x`vxpcrp.
An extra `
Just to confirm that address was right otherwise, I also tried `gaiad keys parse 7CB07B94FD743E2A8520C2B50DA4B03740643BF5`, which produced this:
```
formats:
- cosmos10jc8h98awslz4pfqc26smf9sxaqxgwl4apx3rn
- cosmospub10jc8h98awslz4pfqc26smf9sxaqxgwl4z82z6k
- cosmosvaloper10jc8h98awslz4pfqc26smf9sxaqxgwl4c4jy0q
- cosmosvaloperpub10jc8h98awslz4pfqc26smf9sxaqxgwl42kt8d9
- cosmosvalcons10jc8h98awslz4pfqc26smf9sxaqxgwl4vxpcrp
- cosmosvalconspub10jc8h98awslz4pfqc26smf9sxaqxgwl4mjs085
```
```suggestion
sed -i 's%cosmosvalcons1s0686a68krmr8f46ph6fklw0v8us4gdsm7nhz3%cosmosvalcons10jc8h98awslz4pfqc26smf9sxaqxgwl4xvxpcrp%g' genesis.json
```